lesson16/main.js

```diff
@@ -65,133 +26,159 @@ async function fetchData(url) {
     if (!response.ok) {
       renderStatus(response);
       console.error(`${response.status}:${response.statusText}`);
-    }
-    if (!responseData.length) {
+    } else if (!Object.keys(responseData).length) {
       displayInfo("no data");
+    } else {
+      return responseData;
     }
-    return responseData;
   } catch (error) {
     displayInfo(error);
   } finally {
     removeCircle();
-    backButton.classList.remove("hidden");
   }
 }
 
-async function fetchRenderData(inputNumber) {
-  const responseData = await fetchData(url);
-  if (responseData) {
-    renderData(responseData);
-    console.log(inputNumber); //show inputted number
-  }
+function renderCircle() {
+  const loadingCircle = document.createElement("img");
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "ローディング画像";
+  loadingCircle.id = "loading-circle";
+  tabArea.appendChild(loadingCircle);
 }
 
-openButton.addEventListener("click", () => {
-  resetPrompt();
-  nameBox.value = "";
-  numberBox.value = "";
-  modal.classList.remove("hidden");
-  mask.classList.remove("hidden");
-  openButton.classList.add("hidden");
-  fetchButton.setAttribute("disabled", "true");
-  nameBox.focus();
-});
-
-fetchButton.addEventListener("click", () => {
-  const inputName = nameBox.value;
-  const inputNumber = numberBox.value;
-  fetchRenderData(inputName, inputNumber);
-  modal.classList.add("hidden");
-  mask.classList.add("hidden");
-});
-
-function closeModal() {
-  modal.classList.add("hidden");
-  mask.classList.add("hidden");
-  openButton.classList.remove("hidden");
-}
-
-mask.addEventListener("click", () => {
-  closeModal();
-  resetValidation();
-});
-
-closeButton.addEventListener("click", () => {
-  closeModal();
-  resetValidation();
-});
-
-backButton.addEventListener("click", () => {
-  backButton.classList.add("hidden");
-  openButton.classList.remove("hidden");
-  if (errorMessage) {
-    errorMessage.remove();
-  }
-  while (ul.firstChild) {
-    ul.firstChild.remove();
-  }
-});
+function renderStatus(response) {
+  errorMessage.id = "render-status";
+  errorMessage.textContent = `${response.status}:${response.statusText}`;
+  document.body.appendChild(errorMessage);
+}

```

Comment:
IMO
`statusMessage`に統一する方が良いと思いました
```suggestion
function renderStatusMessage(response) {
  statusMessage.id = "render-status";
 statusMessage.textContent = `${response.status}:${response.statusText}`;
  document.body.appendChild(statusMessage);
}

```